src/Application/Products/Commands/AddProduct/AddProductCommandHandler.cs

```diff
@@ -25,32 +25,26 @@ public AddProductCommandHandler(IProductRepository productRepository, IDateTime
     public async Task<ProductDto> Handle(AddProductCommand request, CancellationToken cancellationToken)
     {
         var productToAdd = request.MapToEntity();
+        var validationResult = productToAdd.Validate();
+
+        if (!validationResult.IsValid)
+        {
+            throw new ValidationException(validationResult.Errors.Adapt<List<ValidationError>>());
+        }
+
         AddAuditableInformation(productToAdd);
+        var product = await _productRepository.Create(productToAdd);
 
-        if (productToAdd.IsValid())
+        try
         {
-            var result = await _productRepository.Create(productToAdd);
-
-            try
-            {
-                await _stockClient.UpdateStock(result.Id, 1);
-            }
-            catch (Exception ex)
-            {
-                _logger.LogError(ex, "Error updating the stock for the product {Id}", result.Id);
-                result.AddError(
-                    "somecode",
-                    "Product added but stock hasn't been updated.",
-                    "Please update the stock later manually using the stocks endpoint. More info on: link to documentation"
-                    );
-            }
-
-            return result.MapToDto();
+            await _stockClient.UpdateStock(product.Id, 1);
         }
-        else
+        catch (Exception ex)
         {
-            throw new ValidationException(productToAdd.Errors.Adapt<List<ValidationError>>());
+            _logger.LogError(ex, "Error updating the stock for the product {Id}", product.Id);
```

Comment:
I see you removed the idea of adding the error to the object.
I can understand your reasons because I had the same concerns (I guess) but I leave it there to bring up the discussion about being able to return something like "warnings" among the normal response when the problem is not "big" enough to throw.
What do you think?

Reply:
In my opinion this dilemma should be solved not in the template, but in the project that uses it, because there are no neutral decisions here.